letter-shop.md

```diff
@@ -0,0 +1,232 @@
+
+In the case of Interledger, the fulfillment is always a 32-byte string, and the condition is the sha256 hash of that string.
+SHA256 is a one-way hash function, so if you know `fulfillment`, then it's easy and quick to calculate `condition = sha256(fulfillment)`, but if you only know the condition, if you only have a million years or less, it's near-impossible to find (i.e., guess) a fulfillment for which `condition = sha256(fulfillment)` would hold. And in practice, we tend to use rollback timeouts that are of course much shorter! :) Because the transfer is *locked* until the recipient produces the correct fulfillment, the condition of the transfer is a *hash* of its fulfillment, and the transfer will *time* out after a while if the recipient doesn't produce the fulfillment to claim the funds, we call this type of conditional transfer a *Hash Time Lock Contract (HTLC)*. We call this a contract between sender an recipient, that is enforced by the ledger. If the sender and the recipient would exchange the condition and the fulfillment directly, that would be an off-ledger transaction, and instead of a Hash Time Lock Contract, we would use the more general term Hash Time Lock Agreement, to indicate that the interaction happens without having the ledger as an arbitor.
+
+## Paying for your letter
+Visit http://localhost:8000. You'll see something like:
+```txt
+Please send an Interledger payment to test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW with condition A6-zI1uIEtjOXMTDoZLtML1xj6YPxBA6yxIQyVh4qhE
+```
+
+Save the following script as `pay.js`:
+```js
+const Plugin = require('ilp-plugin-xrp-escrow')
+const uuid = require('uuid/v4')
+
+const plugin = new Plugin({
+  secret: 'sndb5JDdyWiHZia9zv44zSr2itRy1',
+  account: 'rGtqDAJNTDMLaNNfq1RVYgPT8onFMj19Aj',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+function sendTransfer (obj) {
+  obj.id = uuid()
+  obj.from = plugin.getAccount()
+  // to
+  obj.ledger = plugin.getInfo().prefix
+  // amount
+  obj.ilp = 'AA'
+  obj.noteToSelf = {}
+  // executionCondition
+  obj.expiresAt = new Date(new Date().getTime() + 1000000).toISOString()
+  obj.custom = {}
+  return plugin.sendTransfer(obj)
+}
+
+plugin.connect().then(function () {
+  plugin.on('outgoing_fulfill', function (transferId, fulfillment) {
+    console.log('Got the fulfillment, you paid for your letter! Go get it at http://localhost:8000/' + fulfillment)
+    plugin.disconnect()
+    process.exit()
+  })
+
+  sendTransfer({
+    to: process.argv[2],
+    amount: '1',
+    executionCondition: process.argv[3]
```

Comment:
Nit: The fact that it's pulling these options from the command line is a little bit hard to find since they're buried down here. Maybe put them up at the top in a variable so it's the first thing you see (after the imports)